File: src/_/types/s3.ts
export interface S3Config {
  accessKeyId: string;
  secretAccessKey: string;
  region: string;
  bucket: string;
  endpointUrl: string | null;
}

export interface S3Object {
  key: string;
  displayName: string;
  size: number;
  lastModified: string;
  isFolder: boolean;
  etag: string;
}

export interface S3ListResult {
  objects: S3Object[];
  currentPrefix: string;
}

export interface S3OperationResult {
  success: boolean;
  message: string;
  processed: number;
  failed: number;
}

export interface AwsStorage {
  id: number;
  code: string;
  name: string;
  nameAlias: string;
  subscribe: string;
  isUpload: boolean;
  isDownload: boolean;
  fileOnly: boolean;
  linkAvailable: string[];
  excludeSubscribe: string[];
}

export interface StorageBugFolders {
  storage: AwsStorage;
  bugs: string[];
}

export interface BugFolderItem {
  bugNo: string;
  inSubscribe: boolean;
  lastModified: string;
}

export interface BugFolderTab {
  name: string;
  nameAlias: string;
  items: BugFolderItem[];
}

export interface UploadFileRequest {
  parentName: string;
  name: string;
  localPath: string;
}

export interface ScannedFile {
  parentName: string;
  name: string;
  filePath: string;
  fullPath: string;
  fileSize: number;
}

export interface LocalFileEntry {
  name: string;
  relativePath: string;
  fullPath: string;
  size: number;
}

export interface DeleteUploadedItem {
  awsCd: string;
  bugNo: string;
}

export interface DownloadAvailability {
  downloadAvailable: boolean;
}

export interface DownloadByStorageResult {
  success: boolean;
  message: string;
  processed: number;
  failed: number;
  syncPath: string;
}

export interface DownloadHistorySearchParams {
  fromDate: string;
  toDate: string;
  awsCd: string;
  bugNo: string;
  isMovedAtLocal: boolean;
  isMovedAtS3: boolean;
}

export interface DownloadHistorySearchItem {
  id: number;
  downloadYmd: string;
  awsCd: string;
  awsName: string;
  isMovedAtLocal: boolean;
  bugNo: string;
}

export interface UploadHistorySearchParams {
  fromDate: string;
  toDate: string;
  awsCd: string;
  bugNo: string;
  isMovedAtS3: boolean;
}

export interface UploadHistorySearchItem {
  uploadYmd: string;
  awsName: string;
  isMovedAtS3: boolean;
  bugNo: string;
  attFiles: string;
}

export interface DownloadHistoryItem {
  id: number;
  downloadYmd: string;
  downloadHms: string;
  syncPath: string;
  downloadCount: number;
  isMovedAtLocal: boolean;
  awsName: string;
  awsNameAlias: string;
}

File: src-tauri/src/models/s3.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct S3Config {
    pub access_key_id: String,
    pub secret_access_key: String,
    pub region: String,
    pub bucket: String,
    pub endpoint_url: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AwsStorage {
    pub id: i32,
    pub code: String,
    pub name: String,
    pub name_alias: String,
    pub subscribe: String,
    pub is_upload: bool,
    pub is_download: bool,
    pub file_only: bool,
    pub link_available: Vec<String>,
    pub exclude_subscribe: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UploadFileRequest {
    pub parent_name: String,
    pub name: String,
    pub local_path: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ScannedFile {
    pub parent_name: String,
    pub name: String,
    pub file_path: String,
    pub full_path: String,
    pub file_size: u64,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DeleteUploadedItem {
    pub aws_cd: String,
    pub bug_no: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct S3Object {
    pub key: String,
    pub display_name: String,
    pub size: u64,
    pub last_modified: String,
    pub is_folder: bool,
    pub etag: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct S3ListResult {
    pub objects: Vec<S3Object>,
    pub current_prefix: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct LocalFileEntry {
    pub name: String,
    pub relative_path: String,
    pub full_path: String,
    pub size: u64,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadAvailability {
    pub download_available: bool,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct S3OperationResult {
    pub success: bool,
    pub message: String,
    pub processed: u32,
    pub failed: u32,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadByStorageResult {
    pub success: bool,
    pub message: String,
    pub processed: u32,
    pub failed: u32,
    pub sync_path: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadHistorySearchParams {
    pub from_date: String,
    pub to_date: String,
    pub aws_cd: String,
    pub bug_no: String,
    pub is_moved_at_local: bool,
    pub is_moved_at_s3: bool,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadHistorySearchItem {
    pub id: i32,
    pub download_ymd: String,
    pub aws_cd: String,
    pub aws_name: String,
    pub is_moved_at_local: bool,
    pub bug_no: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct DownloadHistoryItem {
    pub id: i32,
    pub download_ymd: String,
    pub download_hms: String,
    pub sync_path: String,
    pub download_count: i32,
    pub is_moved_at_local: bool,
    pub aws_name: String,
    pub aws_name_alias: String,
}

#[derive(Debug, Clone, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UploadHistorySearchParams {
    pub from_date: String,
    pub to_date: String,
    pub aws_cd: String,
    pub bug_no: String,
    pub is_moved_at_s3: bool,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct StorageBugFolders {
    pub storage: AwsStorage,
    pub bugs: Vec<String>,
}

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BugFolderItem {
    pub bug_no: String,
    pub in_subscribe: bool,
    pub last_modified: String,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct BugFolderTab {
    pub name: String,
    pub name_alias: String,
    pub items: Vec<BugFolderItem>,
}

#[derive(Debug, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct UploadHistorySearchItem {
    pub upload_ymd: String,
    pub aws_name: String,
    pub is_moved_at_s3: bool,
    pub bug_no: String,
    pub att_files: String,
}
